src/Microsoft.Extensions.SecretManager.Tools/Internal/DotNetCoreSdkResolver.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.Extensions.ProjectModel.Internal
+{
+    internal class DotNetCoreSdkResolver
+    {
+        private readonly string _installationPath;
+
+        public DotNetCoreSdkResolver(string installationDir)
+        {
+            _installationPath = installationDir;
+        }
+
+        private IEnumerable<string> Installed
+            => Directory.EnumerateDirectories(Path.Combine(_installationPath, "sdk"));
```

Comment:
Given the fact that we're already tying this expectation to an SDK folder there's no value in taking in the `installationDir`. Lets group all of our questionable "where is msbuild.exe" assumptions in this one class.